meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf

```diff
@@ -22,3 +22,15 @@ UBINIZE_ARGS = "-p 256KiB -m 1 -O 64"
 
 # Needed to skip particular QA checks that don't apply to U-boot's binary.
 INSANE_SKIP_u-boot += "ldflags textrel"
+
+# 16MB for data partition
+MENDER_DATA_PART_SIZE_MB = "16"
+# combined NOR flash is 128MB
+MENDER_STORAGE_TOTAL_SIZE_MB = "128"
+# ??
+MEDNDER_BOOT_PART_SIZE_MB = "1"
+# align to PEB size 256k
+MENDER_PARTITION_ALIGNMENT_KB = "256"
+#MENDER_STORAGE_RESERVED_RAW_SPACE = "524288"
+
+MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET = "0x100000"
```

Comment:
MTD is partitioned like this:
```
1m(u-boot)ro,
1m(u-boot-env)ro,
-(ubi)
```
`meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc` hijacks `CONFIG_ENV_OFFSET` and sets it to the value of `MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET` thus the offset must match the partitioning for u-boot to work. Just for the record, the offset is a it's counted from the beginning of the flash.

The same offset is used in `meta-mender-core/recipes-bsp/u-boot/u-boot-fw-utils-mender.inc` where it's used to generate `fw_env.config`. However, there is a partition defined for uboot's env, and it will appear as /dev/mtd1 covering flash address space 0x100000 to 0x200000. Using the same value there would yield incorrect configuration.

Frankly, I'm not really fond of any of these settings being autoconfigured. In cases other than building for MMC you're more likely to get it wrong and produce something unusable.